### ISSUE TITLE
Running individual tests is broken in Xcode 12.5

### ISSUE THREAD

**Issue by krasich74** [Quick/Quick#1094]
Hi! First of all, thank you for your work, we, as a team of 70+ developers enjoy Quick every day😊

We run our tests on CI and sometimes we come across flaky tests, which we than run once again with xcodebuild -only-testing: and test-without-building. 

Everything was just fine, until recently, when we upgraded to XCode 12.5.1 and Quick 4.0.0 (previous environment was XCode 12.4 and Quick 3.0.0). Since that time, whenever we run xcodebuild -only-testing:"NameOfFailingTestGoesHere" test-without-building test execution succeeds, however number of tests ran is 0(see the attached log at the bottom).

Also we noticed that if we select to run a single test from XCode TestNavigator, it also show success and 0 tests ran. 

We've conducted a little research and found out that running individual tests works just fine with XCode 12.4 and Quick 4.0.0 and Quick 3.0.0

What's more, we've created few tests using pure XCTests, and those can be run individually (probably because Quick uses dynamic tests and XCTest is a static one, though I might be incorrect).

And if we run not individual tests, but individual suites with --only-testing flag everything works as expected!

Hope, we can do some further investigation and solve the issue (or maybe file a bug at Apple). If you need any additional information, please let me know, I'll do my best!

Thanks!

### Environment

List the software versions you're using:

 - Quick: 4.0.0
 - Nimble: 9.1.0
 - Xcode Version: 12.5.1
 - Swift Version: 5.4.2

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - CocoaPods: 1.10.1

```
env NSUnbufferedIO=YES xcodebuild -workspace NameOf.xcworkspace -scheme TestScheme -derivedDataPath ~/Library/Developer/Xcode/DerivedData/ -destination 'platform=iOS Simulator,id=19DD2B8C-EAF4-46F5-AA8F-8988B6E26C86' -parallel-testing-worker-count 10 -maximum-concurrent-test-simulator-destinations 10 -enableCodeCoverage YES -parallel-testing-enabled=NO -only-testing:TestTargetName/TestSuiteName/TestName test-without-building

User defaults from command line:
    IDEDerivedDataPathOverride = /Users/nikitakrasnov/Library/Developer/Xcode/DerivedData/
    IDEMaxParallelTestingSimulators = 10
    IDEPackageSupportUseBuiltinSCM = YES
    IDETestRunOnlyIdentifiers = (
    "TestTargetName/TestSuiteName/TestName"
)
    parallel-testing-enabled = NO

2021-08-26 11:49:49.105 xcodebuild[18807:2841719] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/IDEFrameworks/IDEFrameworks-18212/IDEFoundation/Testing/IDETestRunSpecificationBuilder.m:678
Details:  Failed to compute path to baseline file during test run spec construction: <XCTHTestRunSpecification: 0x7fdbd7eb8f40>
Object:   <IDETestRunSpecificationBuilder>
Method:   +testRunSpecificationsForTestingSpecifiers:scheme:buildables:withBuildParameters:additionalEnvironmentVariables:additionalCommandLineArguments:testRerunPolicy:includeClangProfileParameters:shouldDebugAppExtensions:error:
Thread:   <NSThread: 0x7fdbcc605a70>{number = 1, name = main}
Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
Testing started
Test Suite 'Selected tests' started at 2021-08-25 23:39:53.871
Test Suite 'TestSuiteName' started at 2021-08-25 23:39:53.872
Test Suite 'TestSuiteName' passed at 2021-08-25 23:39:53.872.
	 Executed 0 tests, with 0 failures (0 unexpected) in 0.000 (0.000) seconds
Test Suite 'Selected tests' passed at 2021-08-25 23:39:53.873.
	 Executed 0 tests, with 0 failures (0 unexpected) in 0.000 (0.001) seconds
2021-08-26 11:49:50.350 xcodebuild[18807:2841719] [MT] IDETestOperationsObserverDebug: 1.191 elapsed -- Testing started completed.
2021-08-26 11:49:50.351 xcodebuild[18807:2841719] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2021-08-26 11:49:50.351 xcodebuild[18807:2841719] [MT] IDETestOperationsObserverDebug: 1.191 sec, +1.191 sec -- `end`
```

**Comment by ikesyo** [comment]
Fixed by #1097.

**Comment by jasonm23** [comment]
Seems to be broken in XCode 14.2
